pkg/asset/internal/templates.go

```diff
@@ -1565,4 +1565,349 @@ subjects:
   namespace: kube-system
 `)
 
+var CiliumCfgTemplate = []byte(`apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cilium-config
+  namespace: kube-system
+data:
+  etcd-config: |-
+    ---
+    endpoints:
+    - {{ range $i, $e := .EtcdServers }}{{ if $i }},{{end}}{{ $e }}{{end}}
+{{- if .EtcdUseTLS }}
+    ca-file: '/var/lib/etcd-secrets/etcd-client-ca.crt'
```

Comment:
Not thrilled Cilium still demands etcd client credentials. Flannel and Calico are not given direct access to the cluster's underlying etcd cluster. I'd say block until Cilium addresses it.

Reply:
@dghubble thoughts on using RBAC for this particular etcd connection?

Reply:
Hey Promaethius, I'm excited for Cilium as well, but scoping down the etcd client cert doesn't alleviate the concern. Its the principle. The Kubernetes cluster's etcd is for Kubernetes alone. No other cluster components are given access to it. Not flannel, not Calico, and not Cilium. Cilium can't be special.

We can't lower the bar just for Cilium and as a distro maintainer, the requirement to mint a special etcd client cert just for Cilium seems unreasonable. Cilium could make use of CRDs or a few choice host locations like flannel and Calico have, in order to align with established expectations.

Not sure of the governance of this repo or how I'm the sole reviewer, but I can't approve this.

Reply:
@dghubble I totally understand, thank you for your feedback.
At this point, is it out of the question to introduce new CLI options for a standalone, external ETCD specifically for Cillium? As far as I understand you, the fact that my PR used kubernetes ETCD cluster blocks this.

Reply:
Agree with @dghubble assessment. Direct access to etcd is a slippery slope. I've seen this convention start to erode a bit with aggregated api servers, but personally would not want to advocate for it here.